curated_transformers/tests/tokenization/test_bert_tokenizer.py

```diff
@@ -0,0 +1,243 @@
+import pytest
+import torch
+
+from curated_transformers._compat import has_hf_transformers, transformers
+from curated_transformers.tokenization import PiecesWithIds
+from curated_transformers.tokenization import BertTokenizer
+from curated_transformers.tokenization.bert_tokenizer import BertPreEncoder
+
+from ..util import torch_assertclose
+
+
+@pytest.fixture
+def toy_tokenizer(test_dir):
+    return BertTokenizer.from_files(
+        vocab_path=test_dir / "toy.wordpieces",
+    )
+
+
+@pytest.mark.skipif(not has_hf_transformers, reason="requires huggingface transformers")
+@pytest.mark.slow
+def test_berttokenizer_hf_tokenizer(sample_texts):
+    tokenizer = BertTokenizer.from_hf_hub(name="bert-base-cased")
+    pieces = tokenizer(sample_texts)
+
+    hf_tokenizer = transformers.AutoTokenizer.from_pretrained("bert-base-cased")
+    hf_pieces = hf_tokenizer(sample_texts)
+
+    assert pieces.ids == hf_pieces["input_ids"]
+
+
+def test_bert_toy_tokenizer(toy_tokenizer, short_sample_texts):
+    encoding = toy_tokenizer(short_sample_texts)
+    _check_toy_tokenizer(encoding)
+
+    assert toy_tokenizer.decode(encoding.ids) == [
```

Comment:
It would be nice to have a roundtrip encode/decode test as well (similar to the GPT-NeoX and RoBERTa tokenizer tests), if possible.

Reply:
As discussed internally, I've added round-trip checks to the HF tokenizer comparison.